docs/CODE_GUIDELINES.md

```diff
@@ -339,31 +373,35 @@ void Test(std::string const& a);
 int const * const someIntPointer;
 ```
 
-### 5.4. Initialization
+### 5.5. Initialization
 
 Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
 
 ✅ Good:
 ```cpp
-int x{};
-int* y{};
-CLog::Log("test: {} {}", x, y);
+int x{3};
+int* y{nullptr};
+bool z = false;
+std::string text; // not primitive
+KindOfStruct theStruct{}; // POD structures or structures with uninitalised members must be initialised with empty brackets
+Log::Log("test: {}, {}, {}", x, y, z);
 ```
 
 ❌ Bad:
 ```cpp
 int x; // used uninitialized
 int* y = nullptr; // default-initialization not using {}
-CLog::Log("test: {} {}", x, y);
+bool z{}; // no value explicitly declared for fundamental type, preferable for better reading
+std::string text{}; // has its default constructor
+Log::Log("test: {}, {}, {}", x, y, z);
 ```
 
-In general, prefer the `{}` initializer syntax over alternatives. This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
+We allow initialise variables by using the C++ forms `{}` or `=` or `()`.
 
-✅ Good:
-```cpp
-int x{5};
-int y{x};
-```
+However, we would like to point out some optional suggestions to follow:
+- Prefer the form `{}` to other's, because allow an explicit type checking for possible narrowing conversions.
+- Prefer the form `{}` when initialising a class/struct variable.
```

Comment:
Prefer the `{}` form when initializing...

_(word order plus inconsistent mixing of US/UK spelling otherwise)_